nemo/collections/nlp/data/language_modeling/text_memmap_dataset.py

```diff
@@ -31,11 +33,49 @@
 __idx_suffix__ = 'idx'  # index file suffix
 
 
+def _create_short_hash(string: str) -> str:
+    """Utility function to compute hash of a string.
+
+    This is used to create unique filenames from a full path.
+
+    Args:
+        string: the string to create a hash for.
+    Returns:
+        A short hash of the string.
+    """
+    # Compute the SHA-256 hash of the string
+    sha256_hash = hashlib.sha256(string.encode()).hexdigest()
+
+    # Take a portion of the hash to create a short hash
+    short_hash = sha256_hash[:8]
+
+    return short_hash
+
+
+def _index_base_name(fn: str, index_mapping_dir: str) -> str:
+    """Return base name of index files.
+
+    This returns the base name associated with specified index
+    files. This base name is the base on top of which suffixes
+    like .npy or .info are added.
+
+    Args:
+        fn: filename to get base name for.
+        index_mapping_dir: directory to save the index mapping to.
+                If None, will write to the same folder as the dataset.
+    """
+    if index_mapping_dir:
+        idx_fn = f"{os.path.join(index_mapping_dir, _create_short_hash(fn))}.{__idx_suffix__}"
```

Comment:
How about replacing `_create_short_hash(fn)` with `fn`?
Both are relying on the name of the file, but the latter is more transparent (can easily remove only that cache).
If you do so please also call `os.path.makedirs(os.path.join(index_mapping_dir, os.path.dirname(fn)))` to make sure all required paths are created.

Reply:
Perfect. I would choose transparency over short names in this case (easier debugging and removal).

Reply:
Hi, I reworked the patch as per your suggestion. I also added test cases for `CSVMemMapDataset`.